contrib/endpoints/src/api_manager/context/global_context.h

```diff
@@ -0,0 +1,117 @@
+/* Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef API_MANAGER_CONTEXT_GLOBAL_CONTEXT_H_
+#define API_MANAGER_CONTEXT_GLOBAL_CONTEXT_H_
+
+#include "contrib/endpoints/src/api_manager/auth/certs.h"
+#include "contrib/endpoints/src/api_manager/auth/jwt_cache.h"
+#include "contrib/endpoints/src/api_manager/auth/service_account_token.h"
+#include "contrib/endpoints/src/api_manager/cloud_trace/cloud_trace.h"
+#include "contrib/endpoints/src/api_manager/gce_metadata.h"
+#include "contrib/endpoints/src/api_manager/proto/server_config.pb.h"
+
+namespace google {
+namespace api_manager {
+
+namespace context {
+
+// A global context shared across all services. It stores
+// * env
+// * server_config
+// * service_account_token
+// * certs and jwt_cache
+// * metadata server and fetched data.
+// * cloud trace object.
+class GlobalContext {
+ public:
+  GlobalContext(std::unique_ptr<ApiManagerEnvInterface> env,
+                const std::string &server_config);
+
+  // the env interface.
+  ApiManagerEnvInterface *env() { return env_.get(); }
+
+  // the service account token store
+  auth::ServiceAccountToken *service_account_token() {
+    return &service_account_token_;
+  }
+
+  // certs and jwt_cache
+  auth::Certs &certs() { return certs_; }
+  auth::JwtCache &jwt_cache() { return jwt_cache_; }
```

Comment:
`Certs` and `JwtCache` shouldn't live here, different service have different trusted public keys, thus they are bounded to service configs.